tests/test_datasource.py

```diff
@@ -56,6 +56,20 @@ def test_simple_csv(path):
     ds = DataSource(path("0_0.csv"), reader_kwargs={"encoding": "utf8", "sep": ","})
     assert ds.get_df().shape == (2, 2)
 
+    ds = DataSource(path("0_0.csv"), reader_kwargs={"encoding": "utf8", "sep": ","})
+    assert ds.get_df().shape == (2, 2)
+
+    ds = DataSource(
+        path("0_0.csv"),
+        reader_kwargs={
+            "preview": True,
+            "preview_args": {"nrows": 1, "offset": 0},
```

Comment:
Why not `"preview": {"nrows": 1, "offset": 0}` ?
Is it important to split it into `preview` and `preview_args` ?

Reply:
Not it's not important to have the two, i updated the check with a `bool(preview)` with your recommended payload of the preview as a dictionary !